prac_07/project_management.py

```diff
@@ -0,0 +1,136 @@
+        elif command == "S":
+            file_name = input("File Name: ")
+            save_data(file_name, projects)
+        elif command == "D":
+            display(projects)
+        elif command == "F":
+            date = datetime.strptime(input("Projects that start after date (dd/mm/yyyy): "), '%d/%m/%Y')
+            filter_projects(projects, date)
+        elif command == "A":
+            projects.append(add_project())
+        elif command == "U":
+            projects = update(projects)
+        else:
+            print("Bad Command")
+        menu()
+        command = input(">> ").upper()
+    print("Thank you for using custom-built project management software.")
+
+
+def menu():  # Space at start to make console more readable
+    print("""
+L: Load
+S: Save
+D: Display
+F: Filter (By Date)
+A: Add
+U: Update
+Q: Quit""")
+
+
+def load_data(file_name):
+    file_handler = open(file_name, 'r')
+    file_handler.readline()  # Clears Headers
+    data = []
+    for line in file_handler:
+        parts = line.strip().split("\t")
+        data.append(Project(parts[0], datetime.strptime(parts[1], "%d/%m/%Y"), int(parts[2]), float(parts[3]), int(parts[4])))
+    file_handler.close()
+    return data
+
+
+def save_data(file_name, data):
+    file_handler = open(file_name, 'w')
+    print(HEADER, file=file_handler)
+    for entry in data:
+        print(f"{entry}", file=file_handler)
+    file_handler.close()
+    print(f"Data saved to: {file_name}")
+
+
+def display(projects):
+    print("Incomplete Projects:")
+    incomplete_projects = [project for project in projects if not project.is_complete()]
+    incomplete_projects.sort()
+    for project in incomplete_projects:
+        print_formatted(project)
+    print("Completed Projects:")
+    complete_projects = [project for project in projects if project.is_complete()]
+    complete_projects.sort()
+    for project in complete_projects:
+        print_formatted(project)
+
+
+def filter_projects(projects, date):  # Sort by date
+    filtered_projects = [project for project in projects if project.start_date >= date]
+    filtered_projects.sort(key=lambda x: x.start_date)
+    for project in filtered_projects:
+        print_formatted(project)
+
+
+def add_project():
+    name = input("Project Name: ")
+    date = datetime.strptime(input("Start Date (dd/mm/yyyy): "), "%d/%m/%Y")
+    priority = int(input("Priority: "))
+    cost = float(input("Cost Est: ").strip('$'))
+    percentage = int(input("Percentage Completed: "))
+    project = Project(name, date, priority, cost, percentage)
+    return project
+
+
+def update(projects):
+    for index, project in enumerate(projects):
+        print(index, end='')
+        print_formatted(project)
+    project_choice = input("Project Choice: ")
+    while not in_range(project_choice, len(projects)):
+        print("invalid Selection")
+        project_choice = input("Project Choice: ")
+    project_choice = int(project_choice)
```

Comment:
you could just use int(input()) when getting the choice and you wouldn't need this line